From 87e19a41f5eb96b594d7d11bc1432262b89bf296 Mon Sep 17 00:00:00 2001
From: Nicolas Boichat <drinkcat@google.com>
Date: Tue, 24 May 2016 12:14:41 +0800
Subject: [PATCH 503/513] FROMLIST: automake: egl: Android: Add libEGL
 dependencies

platform_android.c needs to link with cutils, hardware, and sync
libraries, as well as drm_gralloc libs.

Signed-off-by: Nicolas Boichat <drinkcat@google.com>
(am from https://patchwork.freedesktop.org/patch/88731/)

BUG=b:28030733
TEST=Build cyan-cheets

Change-Id: Ibb4e7c974479ea0ad24108e57210b5cb4b2d9537
---
 configure.ac        | 6 ++++++
 src/egl/Makefile.am | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/configure.ac b/configure.ac
index 149478d..ada8711 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1971,6 +1971,12 @@ for plat in $egl_platforms; do
 		;;
 
 	android)
+		AC_CHECK_LIB([cutils],[__android_log_print],[],
+			[AC_MSG_ERROR([cutils library not found])])
+		AC_CHECK_LIB([hardware],[hw_get_module],[],
+			[AC_MSG_ERROR([hardware library not found])])
+		AC_CHECK_LIB([sync],[sync_wait],[],
+			[AC_MSG_ERROR([sync library not found])])
 		;;
 
 	*)
diff --git a/src/egl/Makefile.am b/src/egl/Makefile.am
index e9033b0..6062856 100644
--- a/src/egl/Makefile.am
+++ b/src/egl/Makefile.am
@@ -85,6 +85,9 @@ dri2_backend_FILES += drivers/dri2/platform_surfaceless.c
 endif
 
 if HAVE_EGL_PLATFORM_ANDROID
+libEGL_la_LIBADD += -lcutils -lhardware -lsync
+libEGL_la_LIBADD += $(DRM_GRALLOC_LIBS)
+AM_CFLAGS += $(DRM_GRALLOC_CFLAGS)
 AM_CFLAGS += -DHAVE_ANDROID_PLATFORM
 dri2_backend_FILES += drivers/dri2/platform_android.c
 endif
-- 
2.8.0.rc3.226.g39d4020

